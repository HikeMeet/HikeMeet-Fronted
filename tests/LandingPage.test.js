/*
import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import LandingPage from "../screens/landing/landing-page";

// מדמים את navigation כך שנוכל לוודא שהמעבר קורה
const mockNavigate = jest.fn();
const mockNavigation = { navigate: mockNavigate, goBack: jest.fn() };

describe("LandingPage", () => {
  it("should render correctly (snapshot)", () => {
    const { toJSON } = render(<LandingPage navigation={mockNavigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should navigate to Login page when pressing login button", () => {
    const { getByText } = render(<LandingPage navigation={mockNavigation} />);
    const loginButton = getByText("Login");
    fireEvent.press(loginButton);
    expect(mockNavigate).toHaveBeenCalledWith("Login", {
      toResetPassword: false,
    });
  });

  it("should navigate to Register page when pressing register button", () => {
    const { getByText } = render(<LandingPage navigation={mockNavigation} />);
    const registerButton = getByText("Register");
    fireEvent.press(registerButton);
    expect(mockNavigate).toHaveBeenCalledWith("Register");
  });
});


*/
