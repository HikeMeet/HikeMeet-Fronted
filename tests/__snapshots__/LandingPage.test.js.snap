// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render correctly (snapshot) 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/Logo2.png",
      }
    }
    style={
      {
        "borderRadius": 60,
        "height": 120,
        "marginBottom": 20,
        "width": 120,
      }
    }
  />
  <Text
    style={
      [
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontWeight": "700",
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    HikeMeet
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "center",
        },
        {
          "color": "#4b5563",
        },
        {
          "paddingLeft": 24,
          "paddingRight": 24,
        },
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    Welcome to HikeMeet, where you can connect with fellow hiking enthusiasts.
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#3b82f6",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "marginBottom": 16,
        "opacity": 1,
        "paddingBottom": 12,
        "paddingTop": 12,
        "width": 192,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#fff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "fontWeight": "600",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#d1d5db",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "opacity": 1,
        "paddingBottom": 12,
        "paddingTop": 12,
        "width": 192,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#000",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "fontWeight": "600",
          },
        ]
      }
    >
      Register
    </Text>
  </View>
</View>
`;
